src/workerd/api/basics.c++

```diff
@@ -496,6 +496,23 @@ AbortSignal::AbortSignal(kj::Maybe<kj::Exception> exception,
     flag(flag),
     reason(kj::mv(maybeReason)) {}
 
+kj::Maybe<jsg::JsValue> AbortSignal::getOnAbort(jsg::Lock& js) {
+  return onAbortHandler.map([&](jsg::JsRef<jsg::JsValue>& ref)
+      -> jsg::JsValue { return ref.getHandle(js); });
+}
+
+void AbortSignal::setOnAbort(jsg::Lock& js, jsg::Optional<jsg::JsValue> handler) {
+  // We only want to accept the handler if it's a valid handler... For anything
```

Comment:
I feel like this is the most likely part of this PR to break someone, if anything. Previously people could set `onabort` to any value.

Do we really need this type check?

Reply:
To be spec compliant yes. The entire change potentially breaks someone. I do believe it's highly unlikely that someone would be using a non-event-handler value for this property anyway.

Reply:
To be clear, is this type check specifically necessary to pass the test we're trying to pass?

Reply:
I agree it's not likely.

What I'm saying is that, in my mind, the value of this type check is less than the cost of a compat flag. So if it came down to it and we decided we needed a compat flag for this change, I would be inclined instead to say, just remove the type check and don't bother with a compat flag.

But if we believe no compat flag is needed then, fine.

One specific question, though: `null` is considered an object. Does this code do the right thing when the handler is assigned to `null`? It looks like we'll save the value `null`, rather than actually null out the `kj::Maybe`. Is that OK?

Reply:
If I'm remembering correctly, the c++ `IsObject()` check returns false on `null` so this should just do the right thing. But even if it did store `null`, that would be fine.

Reply:
It would be preferable to stick to the spec here and keep the check in order to stop bad code from seeping in. I think it's unlikely that anyone reads the value back in in the case of this API, except when manually debugging code.

Reply:
I'd say let's go with it as is. *IF* someone is broken, the entire change can be reverted and landed again behind a compat flag.

Reply:
Honestly these arguments are actually making me less comfortable with the change.

> IF someone is broken, the entire change can be reverted and landed again behind a compat flag.

This isn't the right mindset. Breaking people is not OK even if it's temporary, and we certainly cannot assume that a breakage will actually be visible in our metrics or reported to us. We have to be confident that we will not break anyone in the first place.

> It would be preferable to stick to the spec here and keep the check in order to stop bad code from seeping in. I think it's unlikely that anyone reads the value back in in the case of this API, except when manually debugging code.

This seems self-contradictory: If we think people are likely to write such code, we can't introduce the type check because it will break them. If we don't think people are likely to write such code, then the type check doesn't provide value so it doesn't matter if we introduce it or not.

My opinion is that this change won't break anyone -- with the corollary that it doesn't provide any value. I will approve the change as-is because I don't think it actually matters, but my recommendation would be to remove the type check.

Reply:
I totally agree about the mindset. I don't want to speak for James, but I think he meant that he also doesn't believe that this will break anyone and that's why we should land it as is.

In any case, I agree that we should move on.

Reply:
Going ahead with a change that temporarily accepts whatever value is given and logs if it's the wrong type. If we don't get any hits on the new log, we can keep the new type check. If we do get hits on it, then we either drop the check entirely or introduce a compat flag.